src/Features/header/components/CityLanguageSelect/Language/Language.tsx

```diff
@@ -0,0 +1,69 @@
+import { CookieValueTypes, getCookie, setCookie } from 'cookies-next';
+import Link from 'next/link';
+import { ReactElement, useEffect, useState } from 'react';
+
+interface LanguageSelectProps {
+  languageValue: string | undefined;
+}
+
+// eslint-disable-next-line react/prop-types
+export const Language: React.FC<LanguageSelectProps> = ({ languageValue }): ReactElement => {
+  const [isVisible, setIsVisible] = useState<boolean>(false);
+  const [selectedValue, setSelectedValue] = useState<string | undefined>(languageValue);
+
+  useEffect(() => {
+    const cookieValue = getCookie('languageValue') as CookieValueTypes;
```

Comment:
I think you have the exact same function above. If so, create file `CityLanguageSelect.utils` and put this code into a reusable function, then call this function in these 2 useEffects

Reply:
I've tried to do so, but it throws a lot of errors in useEffect.. or I couldn't manage doing it appropriately. So I decided to leave as it is.